gui/default/syncthing/folder/restoreVersionsModalView.html

```diff
@@ -1,12 +1,14 @@
 <modal id="restoreVersions" status="default" icon="fas fa-undo" heading="{{'Restore Versions' | translate}} ({{folderLabel(restoreVersions.folder)}})" large="yes" closeable="yes">
   <div class="modal-body">
-    <span translate ng-if="!restoreVersions.versions && !restoreVersions.errors">Loading data...</span>
-    <div ng-if="restoreVersions.versions">
+    <div ng-if="restoreVersions.versions && !restoreVersions.errors">
       <div id="restoreTree-container">
         <table id="restoreTree">
           <tbody>
             <tr>
-              <td></td>
+              <td>
+                <span ng-if="sizeOf(restoreVersions.versions) > 0" translate>Loading data...</span>
+                <span ng-if="sizeOf(restoreVersions.versions) == 0" translate>There are no file versions to restore.</span>
+              </td>
```

Comment:
I assume this exists for the time until fancytree is initialised, isn't it? What's the point of the conditional, instead of just always showing "loading data" until fancytree is ready?

Reply:
Good question, and the code was remnants of the previous iteration, however I've now added yet another similar `sizeOf` check in the controller, so that the tree code doesn't run at all when there are no versions. This way, the window loads faster, there is no 1970 year in the disabled date filter, and the two conditionals are actually required again (because without the tree code running, the existing table isn't replaced with the tree table).

I hope the explanation is clear… What do you think?

Reply:
Yep makes sense. I don't think I'll remember/be able to figure it out if I ever stumble over this again though, but it wasn't any better before so 🤷